src/lexer.l

```diff
@@ -298,7 +298,7 @@ void includefile()
   std::string fullname = fullpath.generic_string();
 
   filepath.clear();
-  filename_stack.push_back(fullpath);
+  filename_stack.push_back(std::make_shared<fs::path>(fs::path(fullpath.generic_string())));
```

Comment:
Can't you just pass fullpath directly? `std::make_shared<fs::path>(fullpath)`

Reply:
yes we can - I just got paranoid because I got memory access errors, but those where caused by the comment parser that is also using shared pointers.